pages/docs/concepts/asyncapi-document/url-variable.md

```diff
@@ -0,0 +1,96 @@
+---
+title: URL Variables
+weight: 275
+---
+
+[Event-driven architecture](../../tutorials/getting-started/event-driven-architectures) benefits from a defined base URL and rules for URL variables. You can easily manage multiple API endpoints, handling various server configurations and environments.
+
+URL variables are placeholders for values you can replace during runtime. AsyncAPI enables you to construct dynamic URLs with query parameters and additional information while enhancing the flexibility and maintainability of your API specifications.
+
+## Add dynamic variables
+
+To add variables to the URL using AsyncAPI between two servers. You can use the `server.url` and `components.serverVariables` fields to enable reusable variables across multiple servers.
+
+The diagram below depicts how to construct dynamic URLs with query parameters.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style E fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+style F fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[AsyncAPI Document]
+B[Servers Section]
+C[Base URLs]
+D[components.serverVariables Section]
+E[Domain Variable]
+F[Port Variable]
```

Comment:
As I pointed out in my previous review, variables are not just for domains and ports. In some of our examples, we use them for that matter however these are just examples. You can actually use variables in any part of a URL. For instance:

```yaml
servers:
  broker:
    url: 'mqtt://broker.example.com/{env}'
    variables:
      env:
        default: development
        enum:
          - development
          - staging
          - production
```

or:

```yaml
servers:
  broker:
    url: 'ws://ws.example.com?userId={userId}'
    variables:
      userId:
        description: The ID of the user.
```